internal/venice-common/src/main/java/com/linkedin/venice/utils/SharedObjectFactory.java

```diff
@@ -69,10 +72,12 @@ public int size() {
    * managing an object with the identifier, returns 0
    */
   public int getReferenceCount(String identifier) {
-    if (!objectMap.containsKey(identifier)) {
+    // It is possible that this method is not thread-safe, but since this is only used in tests currently, it is okay
```

Comment:
This should be fine even it is used by the regular code path since it won't throw CME regardless of whether the compute is being executed or not.

Reply:
Yeah, it won't give CME, but the issue is that if the upstream code is doing some operation based on this reference count and the reference counts are continuously moving, then there would be jumps in the ref counts. It is non-idiomatic to do this kind of operation, though